packages/redis/test/Adaptor.test.js

```diff
@@ -0,0 +1,334 @@
+    setMockClient({
+      set: async () => 'OK',
+    });
+    const state = { key: 'name', value: 'mtuchi' };
+    const result = await set(
+      s => s.key,
+      s => s.value
+    )(state);
+    expect(result.data).to.eql('OK');
+  });
+  it('should throw an error if key and value is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set()(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if value is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('patient')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if key is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('', 'kwaku')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+});
+
+describe('hset', () => {
+  it('should set fields value for a specified key', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+
+    const state = {};
+    const result = await hset('patient', { name: 'kwaku' })(state);
+    expect(result.data).to.eql(1);
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+    const state = { key: 'patient', value: { name: 'mtuchi' } };
+    const result = await hset(
+      s => s.key,
+      s => s.value
+    )(state);
+    expect(result.data).to.eql(1);
+  });
+  it('should throw an error if key and value is not specified', async () => {
+    setMockClient({
+      hSet: async () => {
+        throw new Error(
+          'TypeError: Cannot convert undefined or null to object'
+        );
+      },
+    });
+
+    const state = {};
+    try {
+      await hset()(state);
+    } catch (error) {
+      expect(error.message).to.eql(
+        'TypeError: Cannot convert undefined or null to object'
+      );
+    }
+  });
+  it('should throw an error if value is not specified', async () => {
+    setMockClient({
+      hSet: async () => {
+        throw new Error(
+          'TypeError: Cannot convert undefined or null to object'
+        );
+      },
+    });
+
+    const state = {};
+    try {
+      await hset('patient')(state);
+    } catch (error) {
+      expect(error.message).to.eql(
+        'TypeError: Cannot convert undefined or null to object'
+      );
+    }
+  });
+  it('should set an empt key if key is not specified', async () => {
```

Comment:
If you pass `''` apparently redis will set an empty key for that value and won't throw an error. Should i remove the test ?

Reply:
Hmm, good question :thinking: 

Let's think about it :detective: 

It's not really our job in these tests to test how the underlying client implementation works. Our job is to ensure that our adaptor is using the client correctly to get the right behaviour.

This test is making explicit the underling client behaviour that if a key is not defined, the key of `''` will be used instead. I suppose that's useful documentation? But if the redis client changes that behaviour in the next release, our test won't start failing, and so it's probably not a very useful test.

If we're worried that someone is going to pass an undefined or empty key, and that this will produce surprising behaviour, we should probably add logic to the adaptor to throw an error and force the user to deal with it. Not a bad idea, it's just the sort of thing an adaptor should be doing.

To be honest, as this is a very early adaptor version which is just a simple wrapper of the library, I would recommend that we do nothing and remove the test. If you pass bad input to the function, you may get surprisingly behaviour and we won't help you. I'm OK with that at the moment.